From e6f4938584939e7cef95d9a1f76f40c3a705fd8f Mon Sep 17 00:00:00 2001
From: Michal Srb <michalsrb@gmail.com>
Date: Sat, 25 Feb 2017 23:36:35 +0200
Subject: [PATCH 11/13] Optimize FcCompareValueList.

The value list from pattern usually contains many elements while the value list from font usually contains just one.
It is faster to make the cycle with less iterations the outer one.

TODO: Add benchmark data.
---
 src/fcmatch.c | 6 ++----
 1 file changed, 2 insertions(+), 4 deletions(-)

diff --git a/src/fcmatch.c b/src/fcmatch.c
index b81616f..be26af5 100644
--- a/src/fcmatch.c
+++ b/src/fcmatch.c
@@ -440,10 +440,9 @@ FcCompareValueList (FcObject	     object,
     best = 1e99;
     bestStrong = 1e99;
     bestWeak = 1e99;
-    j = 0;
-    for (v1 = v1orig; v1; v1 = FcValueListNext(v1))
+    for (v2 = v2orig, k = 0; v2; v2 = FcValueListNext(v2), k++)
     {
-	for (v2 = v2orig, k = 0; v2; v2 = FcValueListNext(v2), k++)
+	for (v1 = v1orig, j = 0; v1; v1 = FcValueListNext(v1), j++)
 	{
 	    v = (match->compare) (&v1->value, &v1->prep_value, &v2->value, &v2->prep_value);
 	    if (v < 0)
@@ -470,7 +469,6 @@ FcCompareValueList (FcObject	     object,
 		    bestWeak = v;
 	    }
 	}
-	j++;
     }
     if (FcDebug () & FC_DBG_MATCHV)
     {
-- 
2.15.1

